presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1100,6 +1107,25 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
                 ImmutableList.of());
     }
 
+    private static <T> TupleDomain<T> columnWiseUnion(Set<T> columns, Iterable<TupleDomain<T>> tupleDomains)
+    {
+        Map<T, Domain> domainsByColumn = new HashMap<>(columns.size());
+
+        for (TupleDomain<T> tupleDomain : tupleDomains) {
+            Optional<Map<T, Domain>> domains = tupleDomain.getDomains();
+            if (domains.isPresent()) {
+                for (Map.Entry<T, Domain> entry : domains.get().entrySet()) {
+                    T column = entry.getKey();
+                    if (columns.contains(column)) {
```

Comment:
actually since this is a very constrained use case, you should probably also assert that every tuple domain contains the exact same 2 columns just to be safe.